lib/xcodeproj/workspace.rb

```diff
@@ -81,14 +130,36 @@ def <<(projpath)
     # @return [Boolean] whether the project is contained in the workspace.
     #
     def include?(file_reference)
-      @file_references.include?(file_reference)
+      file_references.include?(file_reference)
     end
 
     # @return [String] the XML representation of the workspace.
     #
     def to_s
-      contents = file_references.map { |reference| file_reference_xml(reference) }
-      root_xml(contents.join(''))
+      contents = ''
+      stack = []
+      @document.root.each_recursive do |elem|
+        until stack.empty?
+          last = stack.last
+          if last == elem.parent
```

Comment:
Prefer using `break if last == elem.parent` then remove the `else`/`end` to avoid a useless level of indentation